library/src/main/java/com/google/android/exoplayer/util/PlayerControl.java

```diff
@@ -97,9 +103,24 @@ public void pause() {
 
   @Override
   public void seekTo(int timeMillis) {
-    long seekPosition = exoPlayer.getDuration() == ExoPlayer.UNKNOWN_TIME ? 0
-        : Math.min(Math.max(0, timeMillis), getDuration());
+    long seekPosition = (long) timeMillis;
+    if (exoPlayer.getDuration() == ExoPlayer.UNKNOWN_TIME) {
+      if (availableRange != null) {
```

Comment:
Possible bug: if getDuration == UNKNOWN_TIME and availableRange == null, then the seekPosition can in theory be negative when timeMillis is negative, which was impossible before this change.